src/Datadog.Trace/SpanContextPropagator.cs

```diff
@@ -166,6 +164,43 @@ public SpanContext Extract<T>(T carrier, Func<T, string, IEnumerable<string>> ge
             }
         }
 
+        private static TraceId? ParseTraceId<T>(T headers, string headerName)
+            where T : IHeadersCollection
+        {
+            var headerValues = headers.GetValues(headerName);
+            return ParseTraceId(headerValues, headerName);
+        }
+
+        private static TraceId? ParseTraceId<T>(T carrier, Func<T, string, IEnumerable<string>> getter, string headerName)
+        {
+            var headerValues = getter(carrier, headerName).ToList();
+            return ParseTraceId(headerValues, headerName);
+        }
+
+        private static TraceId? ParseTraceId(IEnumerable<string> headerValues, string headerName)
+        {
+            var headerValuesList = headerValues.ToList();
+            foreach (var headerValue in headerValuesList)
+            {
+                try
+                {
+                    var traceId = TraceId.CreateFromString(headerValue);
+                    if (traceId.Equals(TraceId.Zero))
```

Comment:
is there a possibility that a header would really be like that? would it be a valid Trace ID?
what do you think of getting rid of
```
catch (Exception ex) when (ex is ArgumentOutOfRangeException || ex is InvalidOperationException || ex is OverflowException || ex is FormatException)
```
and returning `TraceId.Zero` in `CreateFromString` in case of any exception?
Maybe it would be also possible to get rid of possible exceptions to improve performance?

Reply:
There is a unit test where 0 is passed and it should fail parsing as invalid TraceId. I could move the try/catch to TraceId class as it might be more readable, but I don't see any performance improvements as the exception will have to be thrown and caught anyway.

Reply:
1. Where is this unit test?
2. I think it would make the code more readable and easier to use.

Reply:
1. Datadog.Trace.Tests.HeadersCollectionTests.Extract_InvalidTraceId
2. I will move that.

Reply:
Moved